db/blip_sync_context.go

```diff
@@ -654,11 +655,16 @@ func (bsc *BlipSyncContext) sendNoRev(sender *blip.Sender, docID, revID string,
 }
 
 // Pushes a revision body to the client
-func (bsc *BlipSyncContext) sendRevision(ctx context.Context, sender *blip.Sender, docID, revID string, seq SequenceID, knownRevs map[string]bool, maxHistory int, handleChangesResponseCollection *DatabaseCollectionWithUser, collectionIdx *int, legacyRev bool) error {
+func (bsc *BlipSyncContext) sendRevision(ctx context.Context, sender *blip.Sender, docID, revID string, seq SequenceID, knownRevs map[string]bool, maxHistory int, handleChangesResponseCollection *DatabaseCollectionWithUser, collectionIdx *int, remoteIsLegacyRev bool) error {
 
 	var originalErr error
 	var docRev DocumentRevision
-	if !bsc.useHLV() {
+	var localIsLegacyRev bool
+	// some of this legacy rev handling is due to change pending CBG-4784
+	if !strings.Contains(revID, "@") {
+		localIsLegacyRev = true
+	}
```

Comment:
Can use `base.IsRevTreeID` helper function to simplify this

```suggestion
	localIsLegacyRev := base.IsRevTreeID(revID)
```